webapp/src/pages/AddPlace.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { ChangeEvent, Component, useEffect } from "react";
```

Comment:
Please document all the classes and methos you have in this file, because it is very big and have many things, so the code could be understood faster. And there are also two render methods here, is that correct?

Reply:
There are two different components inside the class, one for the form and another one for the visualization of the images uploaded. That second component has another return function as it should and it is not documented, I guess there's the difficulty understanding it. I'm commenting it right now.

Reply:
The code is finally commented!! Let me know if you have any doubt or if you want me to comment anything in more depth.